### PR TITLE
Some isolation test movement from TINC to ICW and some unused variable warnings fix

### PR DESCRIPTION
**Pull request by ashwinstar**

### CODE REVIEW

File: src/test/regress/sql/psql_gpdb_du.sql [hunk 1]
```diff
@@ -56,3 +56,7 @@ DROP ROLE test_psql_du_e7;
 CREATE ROLE test_psql_du_e8 WITH SUPERUSER CREATEEXTTABLE (type = 'writable', protocol = 'gphdfs');
 \du test_psql_du_e8
 DROP ROLE test_psql_du_e8;
+
+-- The below command tests that psql doesn't segfault when invoked with invalid
+-- PGOPTIONS.
```
Comment (hlinnaka) [inline src/test/regress/sql/psql_gpdb_du.sql]: This is a pretty random place for this test.

File: src/test/isolation2/sql/relation_size_test.sql [hunk 2]
```diff
@@ -0,0 +1,5 @@
+1:drop table if exists relation_size_test_table;
```
Comment (hlinnaka) [inline src/test/isolation2/sql/relation_size_test.sql]: A comment would be nice, to explain briefly what this test tests.

File: src/test/isolation2/expected/drop_rename.out [hunk 3]
```diff
@@ -47,7 +47,7 @@ ALTER
 1:commit;
 COMMIT
 2<:  <... completed>
-ERROR:  tuple concurrently updated (heapam.c:2661)
+ERROR:  relation "t2" does not exist
```
Comment (hlinnaka) [inline src/test/isolation2/expected/drop_rename.out]: The ERROR is different from before. Was that change intentional? Why?

Comment (ashwinstar) [inline src/test/isolation2/expected/drop_rename.out]: Ya the change is intentional as definitely the error seems correct and better than previously answer file reflected. This test was sitting in repository but was never run as part of anything so, not sure when the code change happened since it was coded which modified the behavior to provide proper error message now. Just got lazy to find the rational what changed the behavior as the current error message seemed right.

File: src/test/regress/sql/psql_gpdb_du.sql (hunk 1 above)
Comment (ashwinstar) [inline src/test/regress/sql/psql_gpdb_du.sql]: Yes, agree felt the same while adding :-) Did try to look through where it can be added didn't find any good alternatives maybe then separate sql file for itself.

**Comment by asimrp** [comment]
While at it, can we add a comment to `reader_waits_for_lock.sql` and rename the table from "mpp..." to something else?

File: src/test/isolation2/sql/relation_size_test.sql (hunk 2 above)
Comment (ashwinstar) [inline src/test/isolation2/sql/relation_size_test.sql]: Looking into tests realized its bogus test, not testing anything. It was supposed to test if passed  invalid (dropped tables's oid) function returns NULL instead of ERRORing out. But it doesn't test that currently, actually testing that doesn't even need this fancy isolation test just can have a simple sql `select pg_relation_size(1212121212);` to validate it. So for now just deleting this test file.

PG 9.2 commit will bring in proper fix for it as it fixes all associated functions and not just `pg_relation_size()`

```
commit fa352d662e57fa150158b9cb0a8f127250f8c97f
Author: Heikki Linnakangas <heikki.linnakangas@iki.fi>
Date:   Thu Jan 19 13:06:30 2012 +0200

    Make pg_relation_size() and friends return NULL if the object doesn't exist.

    That avoids errors when the functions are used in queries like "SELECT
    pg_relation_size(oid) FROM pg_class", and a table is dropped concurrently.

    Phil Sorber
```

**Comment by ashwinstar** [comment]
Addressed PR comments and also added one more commit to fix recently reported coverity issue.
